forem/forem
# Async Hamburger Menu Implementation + Caching Optimizations

Closed [this PR ](https://github.com/forem/forem/pull/18760/files) in favor of opening thsi one that deals with the async hamburger and caching optimizations.

## What type of PR is this? (check all applicable)


## Description
This PR sends HTML over an endpoint /async_info/navigation_links and renders it dynamically when we click the hamburger menu. 

It also implements both Rails caching for the sql query as well as edge caching. After some testing from Mac and I, the browser caching looks like its working as well.

## Related Tickets & Documents



- Related Issue #https://github.com/forem/forem/issues/18639

## QA Instructions, Screenshots, Recordings

1. Go to `http://localhost:3000/`
2. Update your screen to be a mobile viewport
3.  Click on the hamburger menu 

Expected behaviour: Make sure that it makes a request to `/async_info/navigation_links` and returns the HTML that gets rendered in the sidebar. 

Caching notes: 

The Fastly caching can only be tested in production. 
We had temporarily deployed to the benhalpern community and checked the caching attributes:
<img width="874" alt="Screenshot 2022-12-13 at 16 48 49" src="https://user-images.githubusercontent.com/2786819/207365543-1fb2b944-6c63-42fc-870d-de1f03a97a86.png">

The browser caching relies on the Fastly caching headers being set and so that can also only be tested in production. We had temporarily deployed to the benhalpern community and checked the caching attributes:
<img width="1512" alt="Screenshot 2022-12-13 at 16 51 43" src="https://user-images.githubusercontent.com/2786819/207366116-e1fb4160-375d-4bf4-b345-a6badbdaa8e5.png">

The Rails caching can be tested by looking at the sql queries being logged and noting in the rails logs when the cache has been hit.


### UI accessibility concerns?

None

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)
